File: backend/src/utils/objects.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug)]
pub struct Blog {
    pub content: String,
    pub date: u64,
    pub description: String,
    pub title: String,
    pub url: String,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct BlogPreview {
    pub title: String,
    pub url: String,
}

#[derive(Serialize, Debug)]
pub struct Error<'a> {
    pub message: &'a str,
}

File: backend/utils/objects.py
from pydantic import BaseModel


class Blog(BaseModel):
    title: str
    url: str
    date: int = None
    content: str

    @staticmethod
    def build(blog_list):
        dict_blogs = []
        for blog in blog_list:
            blog_convert = list(blog)
            format_obj = {}
            blog_order = ["title", "url", "date", "content"]
            for i in range(len(blog_order)):
                format_obj[blog_order[i]] = blog_convert[i]
            dict_blogs.append(format_obj)
        return dict_blogs
